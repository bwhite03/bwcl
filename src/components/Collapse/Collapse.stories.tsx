import React from 'react';
import { Story, Meta } from '@storybook/react';
import Collapse, { CollapseProps } from './Collapse';

export default {
  title: 'Collapse',
  component: Collapse,
} as Meta;

const Template: Story<CollapseProps> = (args) => <Collapse {...args} />;

export const CollapseExample = Template.bind({});
CollapseExample.args = {
  id: 'collapse',
  label: 'collapse',
  children: [<div>Here is some content</div>],
};
