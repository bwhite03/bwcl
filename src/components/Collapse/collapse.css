.wrap-collapse {
  width: 90%;
  margin: 0 auto;
}

.wrap-collapse input[type='checkbox'] {
  display: none;
}

.lbl-toggle-collapse,
.lbl-toggle-collapsed {
  display: block;
  font-weight: bold;
  font-family: monospace;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  padding: 1rem;
  color: #e37a24;
  background: #171620;
  border-radius: 7px;
  transition: all 0.24s ease-out;
  cursor: pointer;
  margin-bottom: 0;
}

.lbl-toggle-collapse:hover {
  color: #7c5a0b;
}

.lbl-toggle-collapse::before {
  content: '';
  display: inline-block;
  border-top: 5px solid transparent;
  border-bottom: 5px solid transparent;
  border-left: 5px solid currentColor;
  vertical-align: middle;
  margin-right: 0.7rem;
  transform: translateY(-2px);
  transition: transform 0.2s ease-out;
}

.toggle-collapsed::before {
  border-bottom: 5px solid transparent;
  border-left: 5px solid;
  border-top: 5px solid transparent;
  content: '';
  display: inline-block;
  margin-right: 0.7rem;
  transform: translateY(-2px);
  transition: transform 0.2s ease-out;
  vertical-align: midde;
}

.collapse-content {
  max-height: 0;
  overflow: hidden;
  transition: max-height 0.5s ease-in-out, padding 0.3s ease-in;
}

.collapse-content.collapsed {
  max-height: 0px;
  background: rgba(67, 71, 97, 0.2);
}

.collapse-content.expanded {
  display: block;
  max-height: 100vh;
  background: rgba(67, 71, 97, 0.2);
  padding: 9px;
}
