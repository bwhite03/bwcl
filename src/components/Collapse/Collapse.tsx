import React, { useState } from 'react';
import './collapse.css';

export interface CollapseProps {
  id: string;
  label: string;
  children: React.ReactNode;
}

const Collapse = (props: CollapseProps) => {
  const [isCollapsed, setCollapse] = useState(true);
  return (
    <div className="wrap-collapse">
      <input id={props.id} type="checkbox" />
      <label
        data-testid={`collapse-${props.id}`}
        className={`${
          isCollapsed ? 'lbl-toggle-collapse' : 'lbl-toggle-collapsed'
        }`}
        onClick={() => setCollapse(!isCollapsed)}
      >
        {props.label}
      </label>
      <div
        className={`collapse-content ${isCollapsed ? 'collapsed' : 'expanded'}`}
      >
        {props.children}
      </div>
    </div>
  );
};

export default Collapse;
