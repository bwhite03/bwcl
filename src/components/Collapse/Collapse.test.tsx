import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Collapse from './Collapse';

describe('Collapse', () => {
  test('should render correctly', () => {
    const { container } = render(
      <Collapse label="c1" id="c1">
        <h2>test</h2>
      </Collapse>
    );
    expect(container).toMatchSnapshot();
  });

  test('should handle click', () => {
    const { container, getByRole, getByTestId } = render(
      <Collapse label="c1" id="c1">
        <h2>test</h2>
      </Collapse>
    );

    const label1 = container.querySelector('label') as HTMLElement;
    fireEvent.click(label1);
  });
});
