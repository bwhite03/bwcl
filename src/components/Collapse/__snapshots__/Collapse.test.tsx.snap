// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse should render correctly 1`] = `
<div>
  <div
    class="wrap-collapse"
  >
    <input
      id="c1"
      type="checkbox"
    />
    <label
      class="lbl-toggle-collapse"
      data-testid="collapse-c1"
    >
      c1
    </label>
    <div
      class="collapse-content collapsed"
    >
      <h2>
        test
      </h2>
    </div>
  </div>
</div>
`;
